### PR TITLE
[RW-6153][risk=no]  Update existing services that read from data_dictionary entry (Data Dictionary)

### PR DESCRIPTION
**Pull request by NehaBroad**
Rather than using the workbench table, use CDR table ds_data_dictionary to get data dictionary information

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/DataSetController.java [hunk 1]
```diff
@@ -463,7 +463,8 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
       throw new BadRequestException("Invalid Domain");
     }
 
-    return ResponseEntity.ok(dataSetService.findDataDictionaryEntry(domainValue, cdrVersion));
+    return ResponseEntity.ok(
+        dataSetService.findDataDictionaryEntry(domainValue, domain, cdrVersion));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/DataSetController.java:467]: These args are confusing. What is the difference between domainValue and domain?

File: api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package org.pmiops.workbench.cdr.dao;
+
+import org.pmiops.workbench.cdr.model.DbDSDataDictionary;
+import org.springframework.data.repository.CrudRepository;
+
+public interface DSDataDictionaryDao extends CrudRepository<DbDSDataDictionary, Long> {
+  DbDSDataDictionary findDbDSDataDictionaryByFieldNameAndDomain(String field_name, String domain);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java:7]: You can shorten this method name.
```suggestion
  DbDSDataDictionary findByFieldNameAndDomain(String field_name, String domain);
```

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java [hunk 3]
```diff
@@ -998,18 +999,21 @@ public void markDirty(ResourceType resourceType, long resourceId) {
...
 
   @Override
-  public DataDictionaryEntry findDataDictionaryEntry(String fieldName, DbCdrVersion cdrVersion) {
-    List<DbDataDictionaryEntry> dataDictionaryEntries =
-        dataDictionaryEntryDao.findByFieldNameAndCdrVersion(fieldName, cdrVersion);
-
-    if (dataDictionaryEntries.isEmpty()) {
+  public DataDictionaryEntry findDataDictionaryEntry(
+      String fieldName, String domain, DbCdrVersion cdrVersion) {
+    CdrVersionContext.setCdrVersionNoCheckAuthDomain(cdrVersion);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java:1004]: This is what controls which cloud CDR we query and needs to be the first thing set in the Controller, so I would move it up.  Also, we should be using the `CdrVersionService` like so -> `cdrVersionService.setCdrVersion(cdrVersionId);`

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java [hunk 4]
```diff
@@ -998,18 +999,21 @@ public void markDirty(ResourceType resourceType, long resourceId) {
...
+    if (dbDSDataDictionary == null) {
       throw new NotFoundException(
           "No Data Dictionary Entry found for domain: "
               + fieldName
               + " cdr version: "
               + cdrVersion);
     }
-    return dataSetMapper.dbModelToClient(dataDictionaryEntries.get(0));
+    DataDictionaryEntry dataDictionaryEntry = dataSetMapper.dbDsModelToClient(dbDSDataDictionary);
+    dataDictionaryEntry.setCdrVersionId(cdrVersion.getCdrVersionId());
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java:1015]: Can we refactor  this to remove the cdrVersionId property from the DataDictionaryEntry client model? The UI should get the cdrVersionId from the workspace and pass that in the API call.

File: api/src/test/java/org/pmiops/workbench/db/dao/DataSetServiceTest.java [hunk 5]
```diff
@@ -488,6 +493,17 @@ public void testFITBITDomainToBigQueryConfig() {
         .contains("GROUP BY PERSON_ID, DATE");
   }
 
+  @Test
+  public void testDataDictionary() {
+    mockDataDictionaryDao();
```
Comment (freemabd) [inline api/src/test/java/org/pmiops/workbench/db/dao/DataSetServiceTest.java:498]: This Test is annotated with `@DataJpaTest` which means it's loading all the tables and using injected daos. Why are we using a mock call when you can query against the database? Also, I would add `@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)` just below `@DataJpaTest` so Spring will clean the context between test methods.

File: api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package org.pmiops.workbench.cdr.dao;
+
+import org.pmiops.workbench.cdr.model.DbDSDataDictionary;
+import org.springframework.data.repository.CrudRepository;
+
+public interface DSDataDictionaryDao extends CrudRepository<DbDSDataDictionary, Long> {
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java:6]: Can we create a test class for dao?

**Review by freemabd — COMMENTED** [review]
Can we also create a test for `getDataDictionaryEntry` in `DataSetControllerTest`?

File: api/src/main/java/org/pmiops/workbench/api/DataSetController.java (hunk 1 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/api/DataSetController.java:467]: domainValue is basically just value. Its from the old code, I'll change it.

Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/api/DataSetController.java]: Done

File: api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java (hunk 2 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java]: Done

File: api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java (hunk 6 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/cdr/dao/DSDataDictionaryDao.java:6]: Done

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java (hunk 3 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java]: Done, moved to controller

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java (hunk 4 above)
Comment (NehaBroad) [reply api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java]: Snce cdrVersion and other attributes like definedTime or getTransformedByRegisteredTierPrivacyMethods were not needed in UI, hence deleted it from the Data Object

File: api/src/test/java/org/pmiops/workbench/db/dao/DataSetServiceTest.java (hunk 5 above)
Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/db/dao/DataSetServiceTest.java]: Done

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java [hunk 7]
```diff
@@ -998,18 +998,19 @@ public void markDirty(ResourceType resourceType, long resourceId) {
   }
 
   @Override
-  public DataDictionaryEntry findDataDictionaryEntry(String fieldName, DbCdrVersion cdrVersion) {
-    List<DbDataDictionaryEntry> dataDictionaryEntries =
-        dataDictionaryEntryDao.findByFieldNameAndCdrVersion(fieldName, cdrVersion);
-
-    if (dataDictionaryEntries.isEmpty()) {
+  public DataDictionaryEntry findDataDictionaryEntry(
+      String fieldName, String domain, DbCdrVersion cdrVersion) {
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java:1002]:
```suggestion
      String fieldName, String domain) {
```

File: api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java [hunk 8]
```diff
@@ -998,18 +998,19 @@ public void markDirty(ResourceType resourceType, long resourceId) {
...
+    DbDSDataDictionary dbDSDataDictionary =
+        dsDataDictionaryDao.findByFieldNameAndDomain(fieldName, domain);
+    if (dbDSDataDictionary == null) {
       throw new NotFoundException(
           "No Data Dictionary Entry found for domain: "
               + fieldName
               + " cdr version: "
               + cdrVersion);
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java:1010]:
```suggestion
              + " domain: "
              + domain);
```

**Review by freemabd — APPROVED** [review]
Couple of comments/suggestions, but looks good.

**Pull request merged**